tests/hw18.spec.js

```diff
@@ -0,0 +1,169 @@
+    test.beforeEach(async ({page})=>{
+        await page.goto('');
+    })
+
+    // Delete test user after each successful sign up
+    test.afterEach(async ({page})=>{
+        await page.getByRole('link', { name: 'Settings' }).click();
+        await page.getByRole('button', { name: 'Remove my account' }).click();
+        await page.getByRole('button', { name: 'Remove' }).click();
+    })
+
+    test('Should be successfully signed up', async ({page})=>{
+        await page.getByRole('button', { name: 'Sign up' }).click();
+        await page.locator('#signupName').click();
+        await page.locator('#signupName').fill('testName');
+        await page.locator('#signupLastName').fill('testLastName');
+        await page.locator('#signupEmail').fill('test-email@test.test');
+        await page.getByLabel('Password', { exact: true }).fill('Qwertyui1');
+        await page.getByLabel('Re-enter password').fill('Qwertyui1');
+        await page.getByRole('button', { name: 'Register' }).click();
+
+        await expect(page.getByRole('button', { name: 'My profile' })).toBeVisible();
+    })
+})
+
+test.describe('Input fields validation', ()=>{
+    test.beforeEach(async ({page})=>{
+        await page.goto('');
+        await page.getByRole('button', { name: 'Sign up' }).click();
+    })
+
+    test.afterEach(async ({page})=> {
+        await expect(page.getByRole('button', { name: 'Register' })).toBeDisabled();
+    })
+
+    test('"Name" input field validation', async ({page})=>{
+        // Empty name field validation
+        await page.locator('#signupName').focus();
+        await page.locator('#signupName').blur();
+        
+        // Border of input field is red
+        await expect(page.locator('#signupName')).toHaveCSS('border-color', 'rgb(220, 53, 69)');
+
+        await expect(page.getByText('Name required')).toBeVisible();
+        await page.locator('#signupName').fill('test');
+        await expect(page.getByText('Name required')).toBeHidden();
+
+        // Invalid name validation
+        await page.locator('#signupName').fill('test123');
+        await expect(page.getByText('Name is invalid')).toBeVisible();
+        await page.locator('#signupName').fill('testName');
+        await expect(page.getByText('Name is invalid')).toBeHidden();
+
+        // Invalid length validation
+        await page.locator('#signupName').fill('t');
+        await expect(page.getByText('Name has to be from 2 to 20 characters long')).toBeVisible();
+        await page.locator('#signupName').fill('te');
+        await expect(page.getByText('Name has to be from 2 to 20 characters long')).toBeHidden();
+        await page.locator('#signupName').fill('test123test123test12');
+        await expect(page.getByText('Name has to be from 2 to 20 characters long')).toBeHidden();
+        await page.locator('#signupName').fill('test123test123test123');
+        await expect(page.getByText('Name has to be from 2 to 20 characters long')).toBeVisible();
+    })
```

Comment:
overcomplicated

you should have either dedicated tests for each validation rule or use test.step & expect.soft

Reply:
Even you separated it be comments to see where each scenario starts

but you can make it clear without comments